src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -238,28 +235,28 @@ export default class ExtensionRequest extends Vue {
       return DateUtilities.addMonthsToDate(19, DateUtilities.dateToYyyyMmDd(this.data.incorporationDate))
     } else {
       // For subsequent AGMs, min shouldn't be later than reference date + 15 months + 1 month
-      return 'placeholder for future work'
+      return DateUtilities.addMonthsToDate(16, this.data.prevAgmDate)
     }
   }
 
-  /**
-   * Whether to disable the editing of AGM Year field.
-   * Editable (false) when Is this first AGM is set to No.
-   * Non-Editable (true) when we first load (null) or Is first AGM is Yes.
-   * Set the agmYear as incorporation date if not null and true.
-   */
+  /** Whether to disable the editing of AGM Year field. */
```

Comment:
Refactored this to only handle enabling/disabling the AGM Year field, and moved logic of setting AGM Year into a watcher on line 253. The point of this is that the line `this.data.agmYear` would prevent the agmYear from getting set if `isFirstAgm == false` while it was inside this getter. Moving it to the watcher resolves this issue.